libparsec/src/cli/invite_device.rs

```diff
@@ -0,0 +1,130 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use clap::Args;
+use std::path::PathBuf;
+use terminal_spinners::{SpinnerBuilder, DOTS};
+
+use libparsec_client_connection::{AuthenticatedCmds, ProxyConfig};
+use libparsec_platform_device_loader::load_device;
+use libparsec_protocol::authenticated_cmds::latest::invite_new::{
+    self, InviteNewRep, UserOrDevice,
+};
+use libparsec_types::{BackendInvitationAddr, DeviceAccessStrategy, InvitationType};
+
+use crate::{cli::*, get_default_config_dir, list_available_devices};
+
+#[derive(Args)]
+pub struct InviteDevice {
+    /// Parsec config directory
+    #[arg(short, long)]
+    config_dir: Option<PathBuf>,
```

Comment:
```Rust
`PathBuf` doesn't implement `std::fmt::Display`
the trait `std::fmt::Display` is not implemented for `PathBuf`
call `.display()` or `.to_string_lossy()` to safely print paths, as they may contain non-Unicode data
required for `PathBuf` to implement `ToString`
```

Reply:
:shrug: 

```suggestion
    #[arg(short, long, default_value = get_default_config_dir().display())]
    config_dir: PathBuf,
```

Reply:
The type would not be `Pathbuf` then :/

Reply:
Clap should use `PathBuf::from_str` with the default value, so every-thing should be allright

Reply:
I don't want an empty pathbuf

Reply:
We do you mean by

> empty `PathBuf`

? 

Clap will use the value provided by `default_value`, If `get_default_config_dir` return an empty path, that's something else

Reply:
I don't understand what you mean, your suggestions don't work

Reply:
```Rust
    #[arg(short, long, default_value_t = get_default_config_dir().display())]
    config_dir: Display,
```
Is not what I want

Reply:
That's not what I suggested:

```suggestion
    #[arg(short, long, default_value = get_default_config_dir().display())]
    config_dir: PathBuf,
```

> Note the `config_dir: PathBuf`

Reply:
```Rust
error[E0277]: the trait bound `clap::builder::OsStr: From<std::path::Display<'_>>` is not satisfied
    --> cli/src/invite_device.rs:18:40
     |
18   |     #[arg(short, long, default_value = get_default_config_dir().display())]
     |                        -------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<std::path::Display<'_>>` is not implemented for `clap::builder::OsStr`
     |                        |
     |                        required by a bound introduced by this call
     |
     = help: the following other types implement trait `From<T>`:
               <clap::builder::OsStr as From<Str>>
               <clap::builder::OsStr as From<&clap::builder::OsStr>>
               <clap::builder::OsStr as From<&Str>>
               <clap::builder::OsStr as From<&&'static std::ffi::OsStr>>
               <clap::builder::OsStr as From<&&'static str>>
               <clap::builder::OsStr as From<&'static std::ffi::OsStr>>
               <clap::builder::OsStr as From<&'static str>>
     = note: required for `std::path::Display<'_>` to implement `Into<clap::builder::OsStr>`
     = note: required for `std::path::Display<'_>` to implement `IntoResettable<clap::builder::OsStr>`
note: required by a bound in `clap::Arg::default_value`
    --> /home/engineer/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.4.11/src/builder/arg.rs:1697:46
     |
1697 |     pub fn default_value(mut self, val: impl IntoResettable<OsStr>) -> Self {
     |                                              ^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Arg::default_value`

For more information about this error, try `rustc --explain E0277`.
```

Reply:
And even if you try
```Rust
#[arg(short, long, default_value_t = get_default_config_dir().display())]
config_dir: PathBuf,
```
```Rust
```error[E0308]: mismatched types
  --> cli/src/invite_device.rs:18:42
   |
18 |     #[arg(short, long, default_value_t = get_default_config_dir().display())]
   |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `PathBuf`, found `Display<'_>`
19 |     config_dir: PathBuf,
   |                 ------- expected due to this
   |
help: try removing the method call
   |
18 -     #[arg(short, long, default_value_t = get_default_config_dir().display())]
18 +     #[arg(short, long, default_value_t = get_default_config_dir())]
   |

error[E0277]: `PathBuf` doesn't implement `std::fmt::Display`
  --> cli/src/invite_device.rs:18:24
   |
18 |     #[arg(short, long, default_value_t = get_default_config_dir().display())]
   |                        ^^^^^^^^^^^^^^^ `PathBuf` cannot be formatted with the default formatter; call `.display()` on it
   |
   = help: the trait `std::fmt::Display` is not implemented for `PathBuf`
   = note: call `.display()` or `.to_string_lossy()` to safely print paths, as they may contain non-Unicode data
   = note: required for `PathBuf` to implement `ToString`

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
```

Reply:
I've recheck the `clap` documentation.

What I wanted was


```suggestion
    #[arg(short, long, default_value_os_t = get_default_config_dir())]
    config_dir: PathBuf,
```

> This should work since `default_value_os_t` doesn't require that the type impl `std::ftm::Display` but `std::convert::Into<OsString>` which `PathBuf` does.

Reply:
Ok, that works :+1: